### DISCUSSION TITLE
Configurable suppression of the "Snapshot Failed" overlay, provide an option to be more tolerant of failed snapshots

### DISCUSSION THREAD

**Discussion by douginoz** [koush/scrypted#952] [Ideas]
In an ideal world all the cameras would work perfectly and snapshots would always work.  But in a perfect world we wouldn't need scrypted!
Currently, many/all of my cameras occasionally timeout when scrypted wants to take a snapshot, resulting in the 'Snapshot Failed' overlay.  As the person setting up and running all this in the house, I'm fine with that.  But the whole point of using scrypted is to allow the cameras to be used by many people in the house, and nobody else is technical.  When they see this 'Snapshot Failed' overlay and a black screen, they are disappointed with the entire home automation setup and it's one more thing to make them use it less rather than more.  
There really isn't any good reason for them to see this black screen and that message, and it is making my job much more difficult because of it.
Instead, can I suggest that 2 things change:
1.  The code be more tolerant of these timeouts.  When they occur, they should be ignored, and the latest valid snapshot image should remain untouched.  This could be a configurable option for each camera in the Snapshot Settings in each camera configuration screen - "Ignore snapshot timeouts"
2. It's still important to know that the camera image is outdated, but this can be done by simply looking at the on-screen clock, or by overlaying the date and time when a valid snapshot was taken, or even overlaying an updating "Image current as of nn minutes ago".  It will then be obvious enough to me that the snapshot is old.  And for other non-technical people, they really don't care if it's old.  The overlay should be optional for those that don't care how old the last image was and don't want text obscuring things.

The normal workflow for people using the webcams is to glance at the camera overview that shows all the cameras.  Then for one they're interested in, they click/tap on it to see the live view.  So even if the snapshot is a bit old, the live view will give them the current view anyway, which is all they want to see anyway.  And if the camera isn't working, that will immediately become apparent.

In summary, I really need these "SNAPSHOT FAILED SOMETHING ISN"T WORKING THIS CAMERA IS CRAP" messages to go away, because it just makes everything look bad.  And while it would be nice to go out and buy PoE 2K best-in-breed webcams, the whole point of using scrypted is to be able to use older, slower, crappy webcams that won't always work perfectly!

**Comment by koush** [comment]
Are you using accessory or bridge mode

**Reply by douginoz** [reply]
I don't know what that means sorry

**Comment by koush** [comment]
screenshot your homekit pairing settings for one of the cameras inside scrypted

**Reply by douginoz** [reply]
![image](https://github.com/koush/scrypted/assets/1946361/9863a5b1-d6cb-40e3-8a3f-a56ca617ca83)

**Comment by koush** [comment]
What camera models?

**Reply by douginoz** [reply]
2 Arlo Pro2 using the Arlo Camera plugin and 3 Eufy indoor 2k in rtsp mode.

**Comment by moebis** [comment]
@koush agree this is a much needed feature. I have 3D printer cameras and sometimes they are off. I rather see the last known good image and not "SNAPSHOT FAILED" on my dashboards. This should be a configurable switch in the snapshot settings for those devices.
